### PR TITLE
vim-rescript v3: Remove `server` and vim commands

### PR DESCRIPTION
**Pull request by aspeddro**
Let's only provides the vim runtime files:
1. `syntax`
2. `indentation`
3. `ftdetect`

New [README.md](https://github.com/aspeddro/vim-rescript/tree/v3#readme)

### CODE REVIEW

File: doc/rescript.txt [hunk 1]
```diff
@@ -15,73 +15,18 @@ Plugin Website: <https://github.com/rescript-lang/vim-rescript>
...
-This plugin provides important features to work on ReScript projects, such as:
+This plugin provides basic features to work on ReScript projects, such as:
 
 - Syntax highlighting
-- Code formatting
-- Type hints
-- Goto Definition
-- Autocompletion (via omnicomplete)
+- Fileype detection
+- Basic automatic indetation
```
Comment (ryyppy) [inline doc/rescript.txt]:
```suggestion
- Basic automatic indentation
```

File: README.md [hunk 2]
```diff
@@ -50,89 +25,68 @@ Plugin 'rescript-lang/vim-rescript'
...
-
-## Using vim-rescript's functionality (no vim-coc, no vim-ale, etc)
+## Setup LSP
 
-(`:h rescript-config`)
+First you need install the language server for ReScript from npm
 
-Vim comes with a set of useful functions that are completely self contained and work with any neovim setup without any plugins:
+> **Note**
+> If you are using [mason.nvim](https://github.com/williamboman/mason.nvim) you can install the ReScript Language Server using the command `MasonInsall rescript-language-server`
```
Comment (ryyppy) [inline README.md]:
```suggestion
> If you are using [mason.nvim](https://github.com/williamboman/mason.nvim) you can install the ReScript Language Server using the command `MasonInstall rescript-language-server`
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,15 @@
 # Changelog
 
-## master
+## 3.0.0
```
Comment (ryyppy) [inline CHANGELOG.md]: It will still be `master` until we tag it as 3.0.0

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+
+**Breaking Changes:**
+
+- Remove Server and Vim Commands
```
Comment (ryyppy) [inline CHANGELOG.md]:
```suggestion
- Remove vendored ReScript LS and remove native Vim Commands (e.g. `:RescriptBuild`, etc)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
+  - The ReScript Language Server will now be shipped as a dedicated npm package [rescript-language-server](https://github.com/rescript-lang/rescript-vscode/tree/master/server#rescript-language-server)
```
Comment (ryyppy) [inline CHANGELOG.md:8]:
```suggestion
  - The ReScript Language Server will now be shipped as a dedicated npm package [rescript-language-server](https://github.com/rescript-lang/rescript-vscode/tree/master/server#rescript-language-server)
  - Migrate to the official language server as described in our README instructions
```

File: README.md [hunk 6]
```diff
@@ -50,89 +25,68 @@ Plugin 'rescript-lang/vim-rescript'
 
 " NeoBundle
 NeoBundle 'rescript-lang/vim-rescript'
+```
 
-" Packer
-use 'rescript-lang/vim-rescript'
+```lua
+-- Lazy.nvim
+{ 'rescript-lang/vim-rescript' }
```
Comment (ryyppy) [inline README.md]: does it make sense to provide the `ft` option as well for lazy loading purposes? (disclaimer: I am not a lazy.vim specialist)

```suggestion
{ 'rescript-lang/vim-rescript', ft="rescript" }
```

Comment (aspeddro) [reply README.md]: yup

**Review by ryyppy — APPROVED** [review]
LGTM

**Pull request merged**